moped-toolbox/populate_geometry_transform/transpose_geometry.py

```diff
@@ -0,0 +1,281 @@
+            print("Project ID: ", record["project_id"])
+            print("Component Name: ", record["component_name"])
+            print("Component Subtype: ", record["component_subtype"])
+            print("Feature ID: ", record["feature_id"])
+            print("Default internal table: ", record["internal_table"])
+            print(str(record["feature_id"]) + ": " + str(feature["geometry"]["type"]))
+            pp.pprint(feature["properties"])
+            pp.pprint(feature["geometry"])
+
+        feature_id = None
+        if (record["component_name"] == "Project Extent - Generic" and
+                str(feature["geometry"]["type"]) == "Point"):
+            # These need special handling, I think this component type will be half intersections, half segments.
+            # It is a genuinely un-geographically typed feature. 🤔
+            
+            # i think this code which is so similar to the blocks around it could be DRYed up
+            # but i don't think it's worth the effort here. see comment below.
+
+            sql = f"""
+            insert into feature_intersections
+            """
+
+            fields = []
+            values = []
+            for key, value in feature["properties"].items():
+
+                # key transformation rules
+                key = remove_leading_underscore(key)
+                key = key.lower()
+                key = 'render_type' if key == 'rendertype' else key
+                key = 'knack_id' if key == 'id' else key
+                key = 'source_layer' if key == 'sourcelayer' else key
+                key = 'intersection_id' if key == 'intersectionid' else key
+
+                fields.append(key)
+                values.append(value)
+            
+            sql += "(" + ",\n".join(fields) + ") values ("
+            sql += ",\n".join(["%s"] * len(values)) + ')'
+            sql += "\nreturning id"
+
+            print(sql, values)
+            result = execute(sql, values, get_result=True)
+            feature_id = result["id"]
+
+            sql = f"""
+            update feature_intersections
+            set geography = ST_Multi(ST_SetSRID(ST_GeomFromGeoJSON(%s), 4326))::geography
+            where id = %s
+            """
+            values = [str(feature["geometry"]), feature_id]
+
+            print(sql, values)
+            execute(sql, values, get_result=False)
+
+            # set the override field for this row
+            sql = "update moped_proj_components set feature_layer_id_override = %s where project_component_id = %s"
+
+            ID_OF_INTERSECTION_TABLE_IN_FEATURE_LAYERS = 5
+            values = [ID_OF_INTERSECTION_TABLE_IN_FEATURE_LAYERS, record["project_component_id"]]
+
+            print(sql, values)
+            execute(sql, values, get_result=False)
+
+        elif ((record["component_name"] == "Sidewalk" and
+                record["component_subtype"] == 'With Curb and Gutter' and 
+                str(feature["geometry"]["type"]) == 'Point')
```

Comment:
This traps feature 170 for project 163 when "Sidewalk / With Curb and Gutter" calls for lines but has a point.

Reply:
boo - i think this an artifact of a known bug we have that allows geometry mismatches following the right sequence of clicks. this is flagged as a test project and i'm going to delete it and its components. maybe we should take this opportunity to not migrate deleted components? i would be happy to see the cruft go away.

Reply:
@johnclary wrote:
> to not migrate deleted components

Hmm, i'm torn. As much as I'd love to see everything in a uniform state, I think deleting soft-deleted records does not align with the intent of having a soft delete system in place.

However, if it's an artifact of testing, and all of the records which can be reached via relations to the subject feature are deleted along with it, then I'm game. I guess put another way, I'm 100% down for deleting features by doing something like `delete from moped_project where ... cascade`, where the cascade reaches down and gets the features.

Reply:
well said. my main concern is that more of these edge cases come about between now and when we run this script. i might take on some data cleanup in the meantime.